Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,199 @@
+        AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // The following code is just a stub while building the SdfAssetBuilder. 
+        // Right now, it just generates an empty default procedural prefab. It still needs to be modified
+        // to create a procedural prefab that contains the correct information in it.
+
+        AZStd::string outputFilename = request.m_sourceFile;
+        AzFramework::StringFunc::Path::ReplaceExtension(outputFilename, "procprefab");
+
+        AZStd::string tempAssetOutputPath;
+        AzFramework::StringFunc::Path::ConstructFull(request.m_tempDirPath.c_str(), outputFilename.c_str(), tempAssetOutputPath, true);
+
+        AZStd::string prefabJson = CreateDefaultProcPrefab(request, response);
+
+        // If no prefab string was generated, then return a failure.
+        if (prefabJson.empty())
+        {
+            response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Failed;
+            return;
+        }
+
+        // Save the prefab to our output directory.
+        rapidjson::Document doc;
+        doc.Parse(prefabJson.c_str());
+        AZ::JsonSerializationUtils::WriteJsonFile(doc, tempAssetOutputPath.c_str());
+
+        // Mark the resulting prefab as a product asset with the "procedural prefab" asset type.
+        AssetBuilderSDK::JobProduct sdfJobProduct;
+        sdfJobProduct.m_productFileName = tempAssetOutputPath;
+        sdfJobProduct.m_productSubID = 0;
+        sdfJobProduct.m_productAssetType = azrtti_typeid<AZ::Prefab::ProceduralPrefabAsset>();
+
+        // Right now, just mark that dependencies are handled because there aren't any to handle.
+        // Once the prefab becomes filled with information, we'll need to do more work here to ensure
+        // that the prefab dependencies are parsed out and hooked up correctly.
+        sdfJobProduct.m_dependenciesHandled = true;
+
+        response.m_outputProducts.push_back(AZStd::move(sdfJobProduct));
+        response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Success;
+    }
+
+    AZStd::string SdfAssetBuilder::CreateDefaultProcPrefab(
+        const AssetBuilderSDK::ProcessJobRequest& request,
+        [[maybe_unused]] AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // The following code is similar to DefaultProceduralPrefabGroup::CreatePrefabGroupManifestUpdates().
+        // It might be possible to refactor the two to have shared logic.
+        // It's not worth investigating until the SdfAssetBuilder is fully functional though, because the code
+        // is likely to change pretty drastically still.
+
+        // Create a prefab name based on the source file name.
+        AZStd::string prefabTemplateName{ request.m_sourceFile };
+        AzFramework::StringFunc::Path::ReplaceExtension(prefabTemplateName, "procprefab");
+        AZ::StringFunc::Replace(prefabTemplateName, "\\", "/"); // the source folder uses forward slash
```

Comment:
The AZ::IO:::Path API can be used for this as well with the `FixedMaxPathStringAsPosix` function combined with the ReplaceExtension function..
```suggestion
        // Create a prefab name based on the source file name.
        AZ::IO::FixedMaxPath prefabTemplateName{ AZ::IO::PathView(request.m_sourceFile).FixedMaxPathStringAsPosix() };
        prefabTemplateName.ReplaceExtension("procprefab");
```